3rdparty/libgcov/patches/GCDAProfiling.c.patch

```diff
@@ -0,0 +1,91 @@
+diff --git a/3rdparty/libgcov/compiler-rt/lib/profile/GCDAProfiling.c b/3rdparty/libgcov/compiler-rt/lib/profile/GCDAProfiling.c
+index cbca36551..3c42b1008 100644
+--- a/3rdparty/libgcov/compiler-rt/lib/profile/GCDAProfiling.c
++++ b/3rdparty/libgcov/compiler-rt/lib/profile/GCDAProfiling.c
+@@ -33,6 +33,7 @@
+ #else
+ #include <sys/mman.h>
+ #include <sys/file.h>
++#include <sys/mount.h>
+ #endif
+ 
+ #if defined(__FreeBSD__) && defined(__i386__)
+@@ -58,6 +59,11 @@ typedef unsigned int uint32_t;
+ typedef unsigned long long uint64_t;
+ #endif
+ 
++/* Beginning of the OE modfication */
++#include "openenclave/bits/module.h"
++#include "openenclave/bits/fs.h"
++/* End of the OE modification */
++
+ #include "InstrProfiling.h"
+ #include "InstrProfilingUtil.h"
+ 
+@@ -255,28 +261,42 @@ static int map_file() {
+   if (file_size == 0)
+     return -1;
+ 
++#if 0  // mmap is not supported by OE.
```

Comment:
I'd recommend following the convention used by the upstream project like:
#if defined(COMPILER_RT_HAS_MMAP)

Same in 47, 61, etc.  It would be good to upstream these changes so we don't need to keep patching.

Reply:
Thanks for the suggestion. Since we currently use older version of the library (llvm-7 in order to match clang-7). I think it would make for sense to the upstream the changes once we start supporting the latest version of clang/llvm.